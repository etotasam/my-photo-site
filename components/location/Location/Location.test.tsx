import React from "react";
import { Location } from ".";
import { render } from "@testing-library/react";
import { cleanup } from "@testing-library/react-hooks";

const image = [
  {
    documentId: "",
    width: 200,
    createAt: new Date(),
    height: 200,
    url: "https://picsum.photos/200",
    filename: "",
    id: "test_1",
  },
  {
    documentId: "",
    width: 200,
    createAt: new Date(),
    height: 200,
    url: "https://picsum.photos/200",
    filename: "",
    id: "test_2",
  },
];

const props = {
  locations: image,
};

afterEach(() => {
  cleanup();
});

describe(`Location`, () => {
  it(`propsで受けたデータがlistでrenderingされているか`, () => {
    const { asFragment, getAllByTestId } = render(<Location {...props} />);
    const renderList = getAllByTestId(`h2`).map((el) => el.textContent);
    const imageList = image.map((el) => {
      const name = el.id.split(`_`)[0];
      return `${name.charAt(0).toUpperCase()}${name.slice(1)}`;
    });
    expect(renderList).toEqual(imageList);
    expect(asFragment()).toMatchSnapshot();
  });
});
