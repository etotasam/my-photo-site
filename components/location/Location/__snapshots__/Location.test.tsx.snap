// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location propsで受けたデータがlistでrenderingされているか 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center"
  >
    <h1
      class="t-under-border text-green-600 mt-5 mx-auto"
    >
      Location
    </h1>
  </div>
  <ul
    class="mt-10 mx-auto"
  >
    <li
      class="w-1/2 md:w-1/5 mb-5 inline-block"
      data-testid="breaked"
    >
      <a
        class="block cursor-pointer relative w-[90%] pt-[45%] mx-auto"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        <div
          style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
        >
          <img
            alt=""
            class="pointer-events-none"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </div>
      </a>
      <h2
        class="font-extralight"
        data-testid="h2"
      >
        Test
      </h2>
    </li>
    <li
      class="w-1/2 md:w-1/5 mb-5 inline-block"
      data-testid="breaked"
    >
      <a
        class="block cursor-pointer relative w-[90%] pt-[45%] mx-auto"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        <div
          style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
        >
          <img
            alt=""
            class="pointer-events-none"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </div>
      </a>
      <h2
        class="font-extralight"
        data-testid="h2"
      >
        Test
      </h2>
    </li>
  </ul>
</DocumentFragment>
`;
